src/python/pants/backend/python/rules/pex_from_targets.py

```diff
@@ -112,18 +112,32 @@ async def pex_from_targets(request: PexFromTargetsRequest, python_setup: PythonS
         python_setup,
     )
 
-    requirements = PexRequirements.create_from_requirement_fields(
-        (
-            tgt[PythonRequirementsField]
-            for tgt in all_targets
-            if tgt.has_field(PythonRequirementsField)
-        ),
-        additional_requirements=request.additional_requirements,
-    )
+    if python_setup.options.use_all_requirements:
```

Comment:
You could merge these. The benefit is that we wouldn't lose support for in-line requirements, which is more consistent: https://www.pantsbuild.org/v2.0/docs/python-third-party-dependencies#defining-inline-requirements

Use a set to de-dupe.

Reply:
That would defeat the caching, no?

Reply:
Yep, for the few cases that you need an requirement that is _not_ already in the global resolve. If the explicitly defined is a subset, then it will still work properly.

https://github.com/pantsbuild/pants/pull/10405#discussion_r457597474 for why I think it's important to be able to still specify requirements not in your single global resolve.

Reply:
Are you proposing parsing the lockfile and adding its requirements to the list, rather than providing it as a file?

Reply:
Indeed, I am.

Reply:
Gotcha